### DISCUSSION TITLE
Table Shadows

### DISCUSSION THREAD

**Discussion by toddbanister** [QuestPDF/QuestPDF#642] [General]
I have been exploring QuestPDF now for about a week and have been amazed that I was able to essentially recreate some fairly complex PDFs that I built in the past. The output is very quick and looks fantastic. The only thing that I seem to be missing in the conversion was table shadows. Previously, I used the following CSS to give the very slight shadow to a table (as it really helps create some dimension against the backgrounds).

{box-shadow: 0 2px 4px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19);}

You can see the final outcome from my previous PDF [here](https://www.dropbox.com/scl/fi/j74r8mk0l0bg4bg56652k/TableShadow.png?rlkey=d3hvrga05a2rbccfrp0rudsmv&dl=0).

Is there any way to add these table shadows in QuestPDf? Possibly via SkiaSharp (which I have not explored at all yet)?

Thank you for this amazing project.

Take care,
TB

**Comment by DawidRobertNowak** [comment]
Hello,

We are closing this thread due to inactivity. We apologize for the delayed response. Thank you for taking the time to reach out and contribute to the project — your feedback helps us improve QuestPDF.

Since this thread was opened, QuestPDF added a native [Shadow](https://www.questpdf.com/api-reference/shadow.html) element that maps directly to a CSS box-shadow. Wrapping your table container in .Shadow(new BoxShadowStyle { ... }) with the Color, Blur, OffsetX, and OffsetY properties produces the same soft drop shadow you had in CSS — no SkiaSharp required.

Should you encounter any further problems, please don't hesitate to open a new issue.
Thank you!
